### PR TITLE
fix: abort search on path if cycle detected

### PR DESCRIPTION
**Pull request by sjvans**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
-  entity.own('_hasPersonalData', () => hasPersonalData)
-  return hasPersonalData
+  if (!entity.own($hasPersonalData)) {
+    if (!entity['@PersonalData.EntitySemantics']) entity.set($hasPersonalData, false)
+    else {
+      // default role to entity name
+      if (entity['@PersonalData.EntitySemantics'] === 'DataSubject' && !entity['@PersonalData.DataSubjectRole'])
+        entity['@PersonalData.DataSubjectRole'] = entity.name.match(/\w+/g).pop()
+      // prettier-ignore
+      const hasPersonalData = !!Object.values(entity.elements).some(element =>
```
Comment (David-Kunz) [inline lib/utils.js:17]: Not part of this change, but in general:
With universal CSN, annotations can also be part of the prototype chain

->

```js
for (const element in entity.elements) {/*...*/}
```

**Review by mariayord — APPROVED** [review]
